packages/tupaia-web/src/api/api.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Tupaia
+ * Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ *
+ */
+import axios from 'axios';
+import FetchError from './fetchError';
+import { getApiUrl } from '../utils';
+
+// withCredentials needs to be set for cookies to save @see https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/withCredentials
+axios.defaults.withCredentials = true;
+
+const timeout = 45 * 1000; // 45 seconds
+
+type RequestParameters = Record<string, any> & {
+  params?: Record<string, any>;
+};
+
+type RequestParametersWithMethod = RequestParameters & {
+  method: 'get' | 'post' | 'put' | 'delete';
+};
+const getRequestOptions = (options: RequestParametersWithMethod) => {
+  const locale = window.location.pathname.split('/')[1];
+  return {
+    ...options,
+    timeout,
+    params: {
+      ...options.params,
+      locale,
+    },
+  };
+};
+
+const request = async (endpoint: string, options: RequestParametersWithMethod) => {
```

Comment:
```suggestion
// Todo: Move api request util to ui-components and allow for mapping to backend request type safety
const request = async (endpoint: string, options: RequestParametersWithMethod) => {
```

Can we add a comment like the above so that it's obvious that this is still a work in progress. thanks